ansible_base/jwt_consumer/common/auth.py

```diff
@@ -0,0 +1,203 @@
+import logging
+from urllib.parse import urljoin, urlparse
+
+import jwt
+import requests
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+
+from ansible_base.lib.utils.settings import get_setting
+
+logger = logging.getLogger("ansible_base.jwt_consumer.common.auth")
+
+
+# These fields are used to both map the user as well as to validate the JWT token
+default_mapped_user_fields = [
+    "first_name",
+    "last_name",
+    "email",
+    "is_superuser",
+    "is_system_auditor",
+]
+
+
+class JWTCommonAuth:
+    def __init__(self, user_fields=default_mapped_user_fields) -> None:
+        self.mapped_user_fields = user_fields
+
+    def parse_jwt_token(self, request):
+        logger.debug("Starting JWT Authentication")
+        token = request.headers.get("X-DAB-JW-TOKEN", None)
+        if not token:
+            logger.info("X-DAB-JW-TOKEN header not set for JWT authentication")
+            return None, None
+        logger.debug(f"Received JWT auth token: {token}")
+
+        jwt_key_setting = get_setting("ANSIBLE_BASE_JWT_KEY")
```

Comment:
Oh, maybe disregard my comment above. This seems to be the setting. Something doesn't seem quite right here though. Maybe we clean up this code after this merges.